### PR TITLE
Add ConstantNaming analyzer

### LINKED ISSUE #125: [Analyzer] Constant names should be all uppercase
This should be fairly easy. It's Node\Stmt\Const_

### PR DESCRIPTION
**Pull request by ddmler**
Resolves #125

### CODE REVIEW

File: src/Analyzer/Pass/Statement/ConstantNaming.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+use PhpParser\Node\Stmt\ClassConst;
+use PHPSA\Analyzer\Pass\AnalyzerPassInterface;
+use PHPSA\Analyzer\Pass\ConfigurablePassInterface;
+use PHPSA\Context;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+
+class ConstantNaming implements ConfigurablePassInterface, AnalyzerPassInterface
+{
+
```
Comment (K-Phoen) [inline src/Analyzer/Pass/Statement/ConstantNaming.php]: Extra line here.

File: src/Analyzer/Pass/Statement/ConstantNaming.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        foreach ($stmt->consts as $const) {
+            if ($const->name != strtoupper($const->name)) {
+                $context->notice(
+                    'constant.naming',
+                    'Constant names should be all uppercase.',
+                    $stmt
+                );
+
+                return true;
```
Comment (K-Phoen) [inline src/Analyzer/Pass/Statement/ConstantNaming.php]: If you return inside the loop, all the constants might not be analyzed. Take the following code for instance, only the first invalid constant will be reported:

```php
class Foo {
    const Bar = 1, Baz = 2;
}
```

File: src/Analyzer/Pass/Statement/ConstantNaming.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    /**
+     * @param ClassConst $stmt
+     * @param Context $context
+     * @return bool
+     */
+    public function pass(ClassConst $stmt, Context $context)
+    {
+        foreach ($stmt->consts as $const) {
+            if ($const->name != strtoupper($const->name)) {
```
Comment (K-Phoen) [inline src/Analyzer/Pass/Statement/ConstantNaming.php]: Why not using a strict equality? (`===`)

File: src/Analyzer/Pass/Statement/ConstantNaming.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        return $treeBuilder;
+    }
+
+    /**
+     * @return array
+     */
+    public function getRegister()
+    {
+        return [
+            \PhpParser\Node\Stmt\ClassConst::class
```
Comment (K-Phoen) [inline src/Analyzer/Pass/Statement/ConstantNaming.php]: Unnecessary FQCN usage here: the ClassConst has been imported through a `use` statement.

File: tests/analyze-fixtures/ConstantNaming.php [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Tests\Compiling\Statements;
+
+class ConstantNaming
+{
+    const UPPERCASE = 2;
+    const Lowercase = 1;
```
Comment (K-Phoen) [inline tests/analyze-fixtures/ConstantNaming.php]: Could you add a case when several constants are declared using a single `const` keyword?

```php
const Foo = 1, Bar = 2;
```

**Comment by ddmler** [comment]
@K-Phoen thanks for review I changed all your points.
I copied the FQCN from another analyzer so i changed it there too.
The return inside the loop would have been a minor problem since the notice would show the whole line and therefore both incorrect constants, but I changed it anyway.
Should we add the wrong constant name to the notice message?

**Comment by K-Phoen** [comment]
> Should we add the wrong constant name to the notice message?

As you said, PHPSA shows the incriminated lines so that would not be really useful, wouldn't it? ¯\_(ツ)_/¯

**Comment by ddmler** [comment]
ping @K-Phoen review issues are fixed can you check again?

**Comment by K-Phoen** [comment]
LGTM :+1:

**Pull request merged**